### ISSUE TITLE
Quick Fix satisfies linter but writes incorrect code

### ISSUE THREAD

**Issue by TheSpydog** [Dart-Code/Dart-Code#2605]
Take this example:

```dart
import 'dart:ffi';
import 'package:ffi/ffi.dart';

class MyStruct extends Struct {
  @Uint32()
  int blah;
}

void main(List<String> arguments) {
  Pointer<MyStruct> a = allocate();
  print("Allocated successfully!");
  free(a);
}
```

This code works as expected, but the linter doesn't like it because `a` is a local variable with an explicit type. In VS Code I see the linter message "Omit type annotations for local variables."

In order to fix this, I used the editor's "Quick Fix" tool, which changed the contents of `main` to the following:
```dart
  var a = allocate();
  print("Allocated successfully!");
  free(a);
```
The linter is happy with this, but when I run the code, it throws the following exception:
```
Unhandled exception:
Invalid argument(s): NativeType does not have a predefined size (@unsized). Unsized NativeTypes do not support [sizeOf] because their size is unknown. Consequently, [allocate], [Pointer.load], [Pointer.store], and [Pointer.elementAt] are not available.
#0      _sizeOf (dart:ffi-patch/ffi_patch.dart:42:56)
#1      sizeOf (dart:ffi-patch/ffi_patch.dart:39:10)
#2      allocate (package:ffi/src/allocation.dart:47:33)
#3      main (file:///<snip>/test.dart:11:11)
```
Which makes complete sense, given that it's trying to allocate an unknown type of native object. My best guess of what it _actually_ wanted is this: `var a = allocate<MyStruct>()`.

I'm not totally sure if the fault rests with the VS Code extension for suggesting an incorrect "fix", or on the linter itself for not recognizing the situation and warning accordingly. I decided to post the issue here rather than in the sdk repo, but I apologize if this isn't the right place!

**Comment by DanTup** [comment]
The SDK is probably the right place for this. I don't know if the issue is in the fix or the lint, however both of those are generated from the SDK code so the fix will likely need to be somewhere over there.

I did search for that error in the SDK repo to see if there was an existing issue, but didn't turn up anything (I did find https://github.com/dart-lang/sdk/issues/38646, but I'm not entirely sure if it's related - I'm not very familiar with ffi or Pointers in Dart).

I would recommend raising it in the SDK repo initially and if that's not the right place (but another dart-lang repo is), someone can probably move the issue to the right place. Thanks!
